io/include/pcl/io/impl/pcd_io.hpp

```diff
@@ -115,6 +115,13 @@ pcl::PCDWriter::writeBinary (const std::string &file_name,
     throw pcl::IOException ("[pcl::PCDWriter::writeBinary] Input point cloud has no data!");
     return (-1);
   }
+
+  if (cloud.width * cloud.height != cloud.points.size ())
```

Comment:
Maybe extract this check into a function called from all writers

Reply:
yeah that would make this quite a bit cleaner.